include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,326 @@
+ * \param  query_value          The value the cigar operation is based on.
+ * \param  extended_cigar Whether to print the extended cigar alphabet or not. See seqan3::cigar_op.
+ * \returns A seqan3::cigar_op representing the alignment operation between the
+ *          two values.
+ *
+ * \details
+ *
+ * \attention Note that a seqan3::cigar_op, of a cigar element describing an
+ *            alignment at a certain position, is always based on one of the
+ *            two sequences in the pairwise alignment. In this case, the resulting
+ *            operation is based on the \p query value.
+ */
+template<typename reference_value_type, typename query_value_type>
+//!\cond
+    requires std::EqualityComparableWith<std::remove_reference_t<reference_value_type>, gap> &&
+             std::EqualityComparableWith<std::remove_reference_t<query_value_type>, gap>
+//!\endcond
+cigar_op compare_aligned_values(reference_value_type const reference_value,
+                                query_value_type const query_value,
+                                bool const extended_cigar)
+{
+    return (reference_value == gap::GAP)
+        ? (query_value == gap::GAP)
+            ? cigar_op::P
+            : cigar_op::I
+        : (query_value == gap::GAP)
+            ? cigar_op::D
+            : (extended_cigar)
+                ? (query_value == reference_value)
+                    ? cigar_op::EQ
+                    : cigar_op::X
+                : cigar_op::M;
+}
+
+/*!\brief Compares two aligned sequence values and returns their seqan3::cigar_op.
+ * \ingroup aligned_sequence
+ * \tparam ref_seq_type   Must model seqan3::aligned_sequence_concept.
+ * \tparam query_seq_type Must model seqan3::aligned_sequence_concept.
+ * \param  alignment      The alignment, represented by a pair of aligned sequences,
+ *                        to be transformed into seqan3::cigar_vector based on the
+ *                        second (query) sequence.
+ * \param  offset_start   The offset at the start of the query sequence indicating soft-clipping.
+ * \param  offset_end     The offset at the end of the query sequence indicating soft-clipping.
+ * \param  extended_cigar Whether to print the extended cigar alphabet or not. See seqan3::cigar_op.
+ * \returns A seqan3::cigar_op representing the alignment operation between the
+ *          two values.
+ *
+ * \details
+ *
+ * \attention Note that seqan3::cigar elements (respectively by their
+ *            seqan3::cigar_op) are always based on one of the two sequences in
+ *            a pairwise alignment. In this case, the resulting cigar_vector is
+ *            based on sequence at the second position of the \p alignment pair.
+ *
+ * Example:
+ * \snippet test/snippet/alphabet/cigar/cigar.cpp get_cigar_vector
+ */
+template<aligned_sequence_concept ref_seq_type, aligned_sequence_concept query_seq_type>
+cigar_vector get_cigar_vector(std::pair<ref_seq_type, query_seq_type> const & alignment,
```

Comment:
Mhmm, I really think we should offer this as a view::to_cigar
Let's consider this use case: 
```cpp
for (auto res : align(seq_rng, config)
{
   //res.alignment would be a zip view over the two aligned sequences.
   bam_record.cigar = res.alignment | to_cigar([soft_clipping_begin{}, soft_clipping_end{}]);
}
```
For the time being, I would make the utility function into `detail` and document it that it will be replaced by a view.